framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/WorkbenchKieServerPersistentScenarioImpl.java

```diff
@@ -34,25 +35,43 @@
 import org.kie.cloud.openshift.deployment.KieServerDeploymentImpl;
 import org.kie.cloud.openshift.deployment.WorkbenchDeploymentImpl;
 import org.kie.cloud.openshift.template.OpenShiftTemplate;
+import org.kie.cloud.openshift.util.SSODeployer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class WorkbenchKieServerPersistentScenarioImpl extends OpenShiftScenario implements WorkbenchKieServerPersistentScenario {
+
     private WorkbenchDeploymentImpl workbenchDeployment;
     private KieServerDeploymentImpl kieServerDeployment;
+    private SSODeployment ssoDeployment;
 
     private Map<String, String> envVariables;
+    private boolean deploySSO;
+    private Map<String, String> ssoEnvVariables;
 
     private static final Logger logger = LoggerFactory.getLogger(WorkbenchKieServerPersistentScenarioImpl.class);
 
-    public WorkbenchKieServerPersistentScenarioImpl(Map<String, String> envVariables) {
+    public WorkbenchKieServerPersistentScenarioImpl(Map<String, String> envVariables, boolean deploySSO, Map<String, String> ssoEnvVariables) {
         this.envVariables = envVariables;
+        this.deploySSO = deploySSO;
+        this.ssoEnvVariables = ssoEnvVariables;
     }
 
     @Override
     public void deploy() {
         super.deploy();
 
+        if (deploySSO) {
+            ssoDeployment = SSODeployer.deploy(project, envVariables, ssoEnvVariables);
+
+            envVariables.put(OpenShiftTemplateConstants.SSO_URL, SSODeployer.createSSOEnvVariable(ssoDeployment.getUrl().toString()));
```

Comment:
This is slightly confusing: You're passing `envVariables` to `SSODeployer.deploy` but after that you're adding more variables into it. Is that ok?

Reply:
Yes, because in SsoDeployer are also created users and passwords (via SsoApi) and we can get these data only from envVariables (deployemts for Workbench and Kie Server are not created yet).
Added variables set connection for Workbench and Kie Server to SSO